test/proof-test.js

```diff
@@ -51,58 +51,64 @@ describe('Proof Tests', function () {
+        } else 
+            return [];
+    }
+
+    function getVerifierParams(configPath, proofPath, publicInputPath) {
+        let public_input = loadPublicInput(path.resolve(__dirname, publicInputPath));
         let params = loadParamsFromFile(path.resolve(__dirname, configPath));
         params['proof'] = fs.readFileSync(path.resolve(__dirname, proofPath), 'utf8');
+        params['public_input'] = public_input;
         return params
     }
 
-
     it("Unified Addition", async function () {
         let configPath = "./data/unified_addition/lambda2.json"
         let proofPath = "./data/unified_addition/lambda2.data"
-        let params = getVerifierParams(configPath,proofPath);
-        await deployments.fixture(['testPlaceholderAPIConsumerFixture', 'unifiedAdditionGateFixture']);
+        let publicInputPath = "./data/unified_addition/public_input.json";
+        let params = getVerifierParams(configPath,proofPath, publicInputPath);
+        await deployments.fixture(['testPlaceholderAPIConsumerFixture', 'unifiedAdditionGateFixture', 'placeholderVerifierFixture']);
 
         let testPlaceholderAPI = await ethers.getContract('TestPlaceholderVerifier');
         let unifiedAdditionGate = await ethers.getContract('UnifiedAdditionGate');
-        await testPlaceholderAPI.verify(params['proof'],params['init_params'], params['columns_rotations'],unifiedAdditionGate.address ,{gasLimit: 30_500_000});
-    });
-
-    it("Merkle Tree Poseidon", async function () {
-        let configPath = "./data/merkle_tree_poseidon/circuit_params.json"
-        let proofPath = "./data/merkle_tree_poseidon/proof.bin"
-        let params = getVerifierParams(configPath,proofPath);
-        await deployments.fixture(['testPlaceholderAPIConsumerFixture', 'merkleTreePoseidonGateFixture']);
-
-        let testPlaceholderAPI = await ethers.getContract('TestPlaceholderVerifier');
-        let merkleTreePosidonGate = await ethers.getContract('MerkleTreePoseidonGate');
-
-        await testPlaceholderAPI.verify(params['proof'],params['init_params'], params['columns_rotations'],merkleTreePosidonGate.address ,{gasLimit: 30_500_000});
+        let placeholderVerifier = await ethers.getContract('PlaceholderVerifier');
+        
+        await testPlaceholderAPI.initialize(placeholderVerifier.address);
+        await testPlaceholderAPI.verify(params['proof'],params['init_params'], params['columns_rotations'], params['public_input'],unifiedAdditionGate.address ,{gasLimit: 30_500_000});
     });
 
     it("Mina Base", async function () {
         let configPath = "./data/mina_base/circuit_params.json"
         let proofPath = "./data/mina_base/proof.bin"
-        let params = getVerifierParams(configPath,proofPath);
-        await deployments.fixture(['testPlaceholderAPIConsumerFixture', 'minaBaseGateFixture']);
+        let publicInputPath = "./data/mina_base/public_input.json";
+        let params = getVerifierParams(configPath,proofPath, publicInputPath);
```

Comment:
I dont think we have public input files for either of these cases

Reply:
Are we testing the public input against the proof passed as arguments for all proofs?

Reply:
I'll prepare public input files for it. It'll be nice large example for testing.

Reply:
> Are we testing the public input against the proof passed as arguments for all proofs?

Yeah. Now public input is optional. But it's temporary. It'll be a necessary parameter for all proofs